examples/move/scratch_off/sources/game.move

```diff
@@ -0,0 +1,256 @@
+
+        let new_store = ConvenienceStore<Asset> {
+            id: object::new(ctx),
+            creator: tx_context::sender(ctx),
+            prize_pool,
+            winning_tickets,
+            losing_tickets_left: max_tickets_issued - winning_ticket_count,
+            winning_tickets_left: winning_ticket_count,       
+            original_ticket_count: max_tickets_issued,
+            tickets_issued: 0, 
+            public_key,
+        };
+
+        // Publically share the object
+        transfer::share_object(new_store);
+        coin
+    }
+
+    /// Initializes a ticket and sends it to someone.
+    /// TODO: decide how to do this besides sending a ticket
+    public fun send_ticket<Asset>(
+        player: address,
+        store: &mut ConvenienceStore<Asset>,
+        ctx: &mut TxContext
+    ) {
+        assert!(store.winning_tickets_left > 0, ENoTicketsLeft);
+        store.tickets_issued = store.tickets_issued + 1;
+        let ticket = Ticket {
+            id: object::new(ctx),
+            convenience_store_id: object::uid_to_inner(&store.id),
+            player
+        };
+        transfer::public_transfer(ticket, player);
+    }
+
+    /// Calls evaluate ticket by adding it as a dynamic field to the store
+    /// We actually have a choice here to make users send tickets to our address and write
+    /// the backend to listen to those, or we can list these objects as a dof in the shared_object
+    public fun evaluate_ticket<Asset>(
+        ticket: Ticket,
+        store: &mut ConvenienceStore<Asset>,
+        ctx: &mut TxContext
+    ): ID {
+        assert!(store.tickets_issued <= store.original_ticket_count, ENoTicketsLeft);
+        let ticket_id = object::uid_to_inner(&ticket.id);
+
+        // Modify the ticket holder even if it got transfered so that the evaluation will
+        // go to the player who sent this ticket
+        ticket.player = tx_context::sender(ctx);
+        dof::add(&mut store.id, ticket_id, ticket);
+        event::emit(NewDrawing<Asset> { 
+            ticket_id,
+            player: tx_context::sender(ctx),
+        });
+        ticket_id
+    }
+
+    /// Reveals whether or not a user has won
+    /// This function is called by the convenience store owner
+    /// and we grab a prize from the store based on the number that was sent.
+    /// We also need to update the prize odds in this function and decrease prizes by 1
+    public fun finish_evaluation<Asset>(
+        ticket_id: ID,
+        bls_sig: vector<u8>,
+        store: &mut ConvenienceStore<Asset>,
+        ctx: &mut TxContext
+    ) {
+        // get the game obj
+        if (!ticket_exists<Asset>(store, ticket_id)) return;
+        let Ticket {
+            id,
+            convenience_store_id: _,
+            player
+        } = dof::remove<ID, Ticket>(&mut store.id, ticket_id);
+
+        // verify BLS sig
+        let msg_vec = object::uid_to_bytes(&id);
+        assert!(
+            bls12381_min_pk_verify(
```

Comment:
Does this portion of the code not handle the owner case since only the user with the private key can finish evaluation?

Reply:
no, everything can work well with both variants, it's just that the `bls12381_min_sig_verify` alternative (or more correctly, preferring small sigs over small pk) is cheaper here

Reply:
What I meant is that this verification function here enforces that only the holder of the private key can sign and call this function. So in a way don't we already handle the case where randoms can call and evaluate this function? Or should I enforce a capability to call this function?

Reply:
I also reverted this back because I couldn't get the backend working with `bls12381_min_sig_verify` Maybe we can hop on a call on monday so I better understand this